antarest/study/web/study_data_blueprint.py

```diff
@@ -1577,4 +1582,337 @@ def set_thermal_form_values(
+
+        Args:
+        - `uuid`: The UUID of the study.
+        - `area_id`: The area ID.
+        - `storage_id`: The ID of the short-term storage.
+        - `ts_name`: The name of the time series to retrieve.
+        - `ts`: The time series matrix to update.
+
+        Permissions:
+        - User must have WRITE permission on the study.
+        """
+        logger.info(
+            f"Update time series for study {uuid} and short-term storage {storage_id}",
+            extra={"user": current_user.id},
+        )
+        params = RequestParameters(user=current_user)
+        study = study_service.check_study_access(
+            uuid, StudyPermissionType.WRITE, params
+        )
+        study_service.st_storage_manager.update_matrix(
+            study, area_id, storage_id, ts_name, ts
+        )
+
+    @bp.get(
+        path="/studies/{uuid}/areas/{area_id}/storages/{storage_id}/validate",
+        tags=[APITag.study_data],
+        summary="Validate all the short-term storage time series",
+    )
+    def validate_st_storage_matrices(
+        uuid: str,
+        area_id: str,
+        storage_id: str,
+        current_user: JWTUser = Depends(auth.get_current_user),
+    ) -> bool:
+        """
+        Validate the consistency of all time series for the given short-term storage.
+
+        Args:
+        - `uuid`: The UUID of the study.
+        - `area_id`: The area ID.
+        - `storage_id`: The ID of the short-term storage.
+
+        Permissions:
+        - User must have READ permission on the study.
+        """
+        logger.info(
+            f"Validating time series for study {uuid} and short-term storage {storage_id}",
+            extra={"user": current_user.id},
+        )
+        params = RequestParameters(user=current_user)
+        study = study_service.check_study_access(
+            uuid, StudyPermissionType.READ, params
+        )
+        return study_service.st_storage_manager.validate_matrices(
+            study, area_id, storage_id
+        )
+
+    @bp.post(
+        path="/studies/{uuid}/areas/{area_id}/storages",
+        tags=[APITag.study_data],
+        summary="Create a new short-term storage in an area",
+    )
+    def create_st_storage(
+        uuid: str,
+        area_id: str,
+        form: STStorageCreateForm,
+        current_user: JWTUser = Depends(auth.get_current_user),
+    ) -> str:
+        """
+        Create a new short-term storage in an area.
+
+        Args:
+        - `uuid`: The UUID of the study.
+        - `area_id`: The area ID.
+        - `form`: The name and the group(PSP_open, PSP_closed, Pondage, Battery, Other1, Other2, Other3, Other4, Other5)
+        of the storage that we want to create.
+
+        Returns: The name of the new storage
+
+        Permissions:
+        - User must have READ/WRITE permission on the study.
+        """
+
+        logger.info(
+            f"Create storage short term from {area_id} for study {uuid}",
```

Comment:
"Create short-term storage from {area_id} for study {uuid}" (short term before storage)